### ISSUE TITLE
Add support for kotlinx @Serializable annotated classes to be considered navigation arguments

### ISSUE THREAD

**Issue by jahongir9779** [raamcosta/compose-destinations#51]
Hi, thanx for your work, nav args are not being generated for @Serializable annotated classess, is it possible to add support for @Serializable from kotlinx.serialization while generating nav args? Or am I doing something wrong...

**Comment by raamcosta** [comment]
Hi @jahongir9779 !

Can you give me more details? What is the error? And how is the annotated Composable function code?

Thanks in advance.

**Comment by jahongir9779** [comment]
The error is: `Too many arguments for public final operator fun invoke()`

```
@Composable
@Destination
fun PersonalDataScreen(
    stages: StageData? = null,
    navController: DestinationsNavigator,
    viewModel: RegistrationViewModel,
) {
...
}
```


```
import kotlinx.serialization.Serializable

@Serializable
data class StageData( 
val code: String,
val name: String,
...
)

```

**Comment by raamcosta** [comment]
Ok thanks!

Yeah so for now, this class will not be considered a navigation argument. Currently navigation arguments have to be one of:
String, Boolean, Int, Float, Long, Parcelable, Serializable and Enums. Read more here: https://github.com/raamcosta/compose-destinations/wiki/Destination-arguments#navigation-arguments

So if you want to pass that as a navigation argument, I would suggest you use the @Parcelize plugin. Check how here:
https://developer.android.com/kotlin/parcelize

Or alternatively, you can just make it implement java `Serializable`. (this is somewhat discouraged in Android, given that Parcelable have better performance)

Later on, I might improve the library to support kotlinx serializable, but that will be a low priority enhancement given that we have alternatives.

**Comment by jahongir9779** [comment]
sounds good, thanx)

**Comment by Aidanvii7** [comment]
This would be nice actually, Parcelable currently doesn't support inline/value classes as properties which is a real pain. Kotlin serialisation handles this just fine on the other hand.

**Comment by raamcosta** [comment]
Just merged this support, version 1.4.3 will already contain this support. @Aidanvii7 , @jahongir9779 , if you can test this, please let me know if you find any issue!
